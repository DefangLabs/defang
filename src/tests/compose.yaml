name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: "${COMPOSE_PROJECT_NAME}-dfnx.prod1.defang.dev"
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
