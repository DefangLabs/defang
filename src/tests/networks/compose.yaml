services:
  service1:
    image: example
    networks:
      invalid-network-name: {}
  service2:
    image: example
    networks:
      public: {}
  service3:
    image: example
    networks:
      - public
  service4:
    image: example
    networks:
      - private
networks:
  public:
    external: true
