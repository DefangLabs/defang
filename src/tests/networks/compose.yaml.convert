[
  {
    "name": "service1",
    "platform": 2,
    "internal": true,
    "networks": 1
  },
  {
    "name": "service2",
    "platform": 2,
    "networks": 2
  },
  {
    "name": "service3",
    "platform": 2,
    "networks": 2
  },
  {
    "name": "service4",
    "platform": 2,
    "internal": true,
    "networks": 1
  }
]