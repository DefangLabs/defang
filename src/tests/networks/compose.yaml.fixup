[
  {
    "command": null,
    "entrypoint": null,
    "image": "example",
    "networks": {
      "invalid-network-name": {}
    }
  },
  {
    "command": null,
    "entrypoint": null,
    "image": "example",
    "networks": {
      "public": {}
    }
  },
  {
    "command": null,
    "entrypoint": null,
    "image": "example",
    "networks": {
      "public": null
    }
  },
  {
    "command": null,
    "entrypoint": null,
    "image": "example",
    "networks": {
      "private": null
    }
  }
]