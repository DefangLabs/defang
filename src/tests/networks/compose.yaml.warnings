level=warning msg="service \"service1\": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)"
level=warning msg="service \"service1\": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors"
level=warning msg="service \"service1\": network \"invalid-network-name\" is not defined in the top-level networks section"
level=warning msg="service \"service2\": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)"
level=warning msg="service \"service2\": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors"
level=warning msg="service \"service3\": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)"
level=warning msg="service \"service3\": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors"
level=warning msg="service \"service4\": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)"
level=warning msg="service \"service4\": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors"
level=warning msg="service \"service4\": network \"private\" is not defined in the top-level networks section"