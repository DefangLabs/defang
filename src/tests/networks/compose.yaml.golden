name: networks
services:
  service1:
    networks:
      invalid-network-name: {}
  service2:
    networks:
      public: {}
  service3:
    networks:
      public: null
  service4:
    networks:
      private: null
networks:
  public:
    name: public
    external: true
